### PR TITLE
Fixes bug with indexes when generating the TileJSON for Named Maps.

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
I spent some time to (hopefully) clarify the code a bit and added some more tests to (hopefully) avoid more issues with indexes & infowindows :)

@xavijam can you please CR? Thanks!

### CODE REVIEW

File: src/geo/cartodb-layer-group-base.js [hunk 1]
```diff
@@ -31,36 +31,51 @@ var CartoDBLayerGroupBase = Backbone.Model.extend({
...
+        tiles: urls.tiles,
+        formatter: function (options, data) { return data; }
+      };
     }
+  },
 
-    return {
-      tilejson: '2.0.0',
-      scheme: 'xyz',
-      grids: urls.grids[this._getIndexOfVisibleLayer(layerIndex)],
```
Comment (alonsogarciapablo) [inline src/geo/cartodb-layer-group-base.js]: @xavijam: the problem was basically that this method was returning the wrong index. `urls.grids` is an array that contains one entry for each of the visible layers. In the case of Named Maps `this._getIndexOfVisibleLayer` was taking into account that Named Maps have a tiled layer at the "bottom" and the wrong grid was being included in the TileJSON. The new method `_convertToMapnikLayerIndex` only takes into account visible "mapnik" layers, which will (hopefully) match the index for the given `layerIndex` in `url.grids`.

File: src/geo/cartodb-layer-group-anonymous-map.js [hunk 2]
```diff
@@ -5,10 +5,7 @@ var CartoDBLayerGroupAnonymousMap = CartoDBLayerGroupBase.extend({
     type: 'layergroup'
   },
 
-  // Returns the index of the CartoDB layer in relation to all layers when the map is
-  // an "Anonymous Map". For example, if there are two CartoDB layers and layer #0 is
-  // hidden, this method would return -1 for #0 and 0 for layer #1.
-  _getIndexOfVisibleLayer: function (layerIndex) {
+  _convertToWindshaftLayerIndex: function (layerIndex) {
```
Comment (xavijam) [inline src/geo/cartodb-layer-group-anonymous-map.js]: key function!

**Comment by xavijam** [comment]
:clap: :clap:

**Comment by xavijam** [comment]
Go ahead! Infowindows will never break again :smile_cat:

**Pull request merged**